static/src/javascripts/projects/common/utils/detect.js

```diff
@@ -252,15 +254,20 @@ define([
     }
 
     function getViewport() {
-        var w = window,
-            d = document,
-            e = d.documentElement,
-            g = d.getElementsByTagName('body')[0];
+        var docEl,
+            body;
 
-        return {
-            width:  w.innerWidth  || e.clientWidth  || g.clientWidth,
-            height: w.innerHeight || e.clientHeight || g.clientHeight
-        };
+        if (!viewportCache) {
+            docEl = document.documentElement;
+            body = document.getElementsByTagName('body')[0];
+
+            viewportCache = {
+                width:  window.innerWidth  || docEl.clientWidth  || body.clientWidth,
+                height: window.innerHeight || docEl.clientHeight || body.clientHeight
+            };
+        }
+
+        return viewportCache;
     }
```

Comment:
Thought it be better to minimise viewport calculations for the 99.99% users who _don't_ resize...